src/jsn.erl

```diff
@@ -656,6 +687,46 @@ from_proplist(X, _Options) -> X.
 %% internal functions
 %%==============================================================================
 
+-spec apply_conditions(conditions(), json_term()) -> boolean().
+apply_conditions([Condition|Rest], Element) ->
+    case apply_condition(Condition, Element) of
+        true ->
+            apply_conditions(Rest, Element);
+        false ->
+            false
+    end;
+apply_conditions([], _Element) ->
+    true;
+apply_conditions(Condition, Element) ->
+    apply_condition(Condition, Element).
+
+-spec apply_condition(condition(), json_term()) -> boolean().
+apply_condition({Path, ConditionFun}, Element) when is_function(ConditionFun, 1) ->
+    ConditionFun(get(Path, Element));
+apply_condition({Path, Value}, Element) ->
+    get(Path, Element) == Value;
+apply_condition(ConditionFun, Element) when is_function(ConditionFun, 1) ->
+    ConditionFun(Element);
+apply_condition(_, _) ->
+    erlang:error(invalid_condition).
```

Comment:
can you make this: 

```erlang
apply_condition(Condition, Element) ->
    erlang:error(badarg, [Condition, Element]).
```

It seems like adding the `Args` as the second argument to `erlang:error(...)` is the key to getting a nice error output that is easy to understant. When I do that, I get output like this:

```erlang
jsn:select([{value, <<"username">>}], [fuuu], [jsn:new()]).
** exception error: bad argument
     in function  jsn:apply_condition/2
        called as jsn:apply_condition(fuuu,[])
     in call from jsn:apply_conditions/2 (jsn/_build/default/lib/jsn/src/jsn.erl, line 692)
     in call from jsn:'-select/3-fun-0-'/3 (jsn/_build/default/lib/jsn/src/jsn.erl, line 195)
     in call from lists:filtermap/2 (lists.erl, line 1317)
```